lib/msf/base/simple/exploit.rb

```diff
@@ -149,14 +149,14 @@ def self.exploit_simple(oexploit, opts, &block)
       # Propagate this back to the caller for console mgmt
       oexploit.job_id = exploit.job_id
     rescue ::Interrupt
-      exploit.error = $!
+      oexploit.error = $!
```

Comment:
iirc I think `exploit.error` is important here so the caller can keep track of errors on the replicant instance that gets yielded here: 

https://github.com/rapid7/metasploit-framework/blob/d626886250f66f745bcf2f0030f8b1e0b8575034/lib/msf/base/simple/exploit.rb#L64

Reply:
Was there a reason for swapping this over to `oexploit` 👀

Reply:
Yes because without switching it to `oexploit`, the error is set on the replicant instead of the reference the caller owns. This means the caller can't get the error information which is now needed here:

https://github.com/rapid7/metasploit-framework/blob/678821ca35b16f6be9d58a24467d15f4945d46d9/lib/msf/ui/console/command_dispatcher/exploit.rb#L207

Now I haven't tested this, but I believe that the exception handling here would be executed after the block so if Pro raises its exception and thus has concluded. It seems like Pro should be getting the exception from the module it's passing rather than storing a reference to the one passed to it via the block.

Or we can just set the error on both ala `oexploit.error = exploit.error = $!`. That'd work too.

Reply:
> This means the caller can't get the error information which is now needed here:

So I believe you can use the yielded replicant exploit from the command dispatcher caller, that's the pattern to follow as far as I understand

https://github.com/rapid7/metasploit-framework/blob/d626886250f66f745bcf2f0030f8b1e0b8575034/lib/msf/base/simple/exploit.rb#L64

I'm happy to sync up on a call if I'm misunderstanding something though :100: